x/gamm/types/account.go

```diff
@@ -25,17 +25,17 @@ type PoolAccountI interface {
 	GetTotalShare() sdk.Coin
 	AddTotalShare(amt sdk.Int)
 	SubTotalShare(amt sdk.Int)
-	AddRecords(records []Record) error
-	GetRecord(denom string) (Record, error)
-	SetRecord(denom string, record Record) error
-	GetRecords(denoms ...string) ([]Record, error)
-	SetRecords(record []Record) error
-	GetAllRecords() []Record
+	AddPoolAssets(PoolAssets []PoolAsset) error
+	GetPoolAsset(denom string) (PoolAsset, error)
+	SetPoolAsset(denom string, PoolAsset PoolAsset) error
+	GetPoolAssets(denoms ...string) ([]PoolAsset, error)
+	SetPoolAssets(PoolAsset []PoolAsset) error
+	GetAllPoolAssets() []PoolAsset
```

Comment:
I do wonder if we should rename these methods to remove the `Pool` prefix.

Reply:
Technically PoolAsset includes the weight, and is a bit different from the typical notion of 'assets' so do we really need to change it? But then again, I'm going to assume whoever is looking at the code would know it. So I'm fine either way.

Reply:
Ok, sounds good. I do think theres probably a better name than `PoolAsset` as well, since it also includes the weight, but I think this name is significantly more descriptive